locations/spiders/mcdonalds_lu.py

```diff
@@ -1,119 +1,46 @@
-                }
-
-            index = index + 1
-
-        day_groups.append(this_day_group)
-
-        if not day_groups:
-            return None
-        opening_hours = ""
-        if len(day_groups) == 1 and day_groups[0]["hours"] in (
-            "00:00-23:59",
-            "00:00-00:00",
-        ):
-            opening_hours = "24/7"
-        else:
-            for day_group in day_groups:
-                if day_group["from_day"] == day_group["to_day"]:
-                    opening_hours += "{from_day} {hours}; ".format(**day_group)
-                else:
-                    opening_hours += "{from_day}-{to_day} {hours}; ".format(**day_group)
-            opening_hours = opening_hours[:-2]
-
-        return opening_hours
-
-    def parse_name(self, name):
-        match = re.search(r"<h1>(.*)</h1>", name)
-        return match.groups()[0].strip()
+    allowed_domains = ["mcd.lu"]
+    start_urls = ["https://mcd.lu/"]
 
     def parse_latlon(self, data):
         match = re.search(r"sll=(.*),(.*)&amp;ss", data)
-        return match.groups()[0].strip(), match.groups()[1].strip()
+        if match:
+            return match.groups()[0].strip(), match.groups()[1].strip()
+        else:
+            return None, None
 
     def parse_phone(self, phone):
-        match = re.search(r"Tel(.*)<br", phone)
-        return match.groups()[0].strip()
+        match = re.search(r'Tel <span itemprop="telephone" content="(.*)">', phone)
+        if match:
+            return match.groups()[0].strip()
+        else:
+            return None
 
     def parse_address(self, address):
         address = address[address.find("<h2>Adresse</h2>") + 16 : address.find("Tel")]
         match = re.sub("<[^<]+?>", "", address)
         return " ".join(match.split())
 
-    def parse_store(self, response):
-        data = response.text
-
-        name = self.parse_name(data)
-        address = self.parse_address(data)
-        phone = self.parse_phone(data)
-        lat, lon = self.parse_latlon(data)
-        properties = {
-            "ref": response.meta["ref"],
-            "phone": phone,
-            "lon": lon,
-            "lat": lat,
-            "name": name,
-            "addr_full": address,
-        }
-        opening_hours = self.store_hours(response)
-        if opening_hours:
-            properties["opening_hours"] = opening_hours
+    def parse(self, response, **kwargs):
+        for ref in response.xpath('//a[starts-with(@id, "snav_1_")]/@id').getall():
+            yield Request(
+                "https://mcd.lu/content.php?r={}&lang=de".format(ref.replace("snav_", "")), callback=self.parse_sd
+            )
 
-        yield Feature(**properties)
+    def post_process_item(self, item, response, ld_data, **kwargs):
+        item["website"] = None
+        item["addr_full"] = self.parse_address(response.text)
+        item["phone"] = self.parse_phone(response.text)
+        item["lat"], item["lon"] = self.parse_latlon(response.text)
```

Comment:
I looked at the extract_google and it does not work because the iframe does not have maps/embed in the source.

Reply:
I think I found the problem.

In `locations/google_url.py` find the line `url = url.replace("google.co.uk", "google.com")` and beneath it, add a replacement for `google.de`.

Reply:
Not quite, the issue was that the `yield from response.xpath('//iframe[contains(@src, "maps/embed")]/@src').getall()`. It does not collect the link from the iFrame because the links do not have `maps/embed` path in them. It is just `maps.google.de/maps?f=q&amp....`. It does work if I add another `yield from` line with just `@src, "maps"`.

Also in a couple of them the the iframe node is commented out of the HTML. Thus the google extract will not work but using regex will still work.

Reply:
Thanks for investigating. The fact they sometimes comment out the `iframe` element with the embedded Google Maps `@src` is a compelling reason for searching response.text() for the `sll=` attribute to Google Maps.